src/common/datomish/transact.cljc

```diff
@@ -45,9 +45,11 @@
     [conn]
     "Get the full transaction history DB associated with this connection."))
 
-(defrecord Connection [current-db]
+(defrecord Connection [current-db transact-chan]
   IConnection
-  (close [conn] (db/close-db @(:current-db conn)))
+  (close [conn]
+    (a/close! (:transact-chan conn))
+    (db/close-db @(:current-db conn)))
```

Comment:
The transact chan might still have items, and those will be processed… and will try to use the DB.

I think you should:

- Optionally: empty `transact-chan`, putting a "database closed before you ran, sorry" error into each task's result chan. If you don't, existing queued transactions will run before we close.
- Then, either:
  - Immediately close the DB, hoping that the current transact will throw and report interruption, or
  - Push a "close the DB" sentinel into `transact-chan` just before closing it. Closing the DB will then occur inside the transactor loop, between transacts. (When you close the DB, check again for tasks waiting behind, and tell them 'sorry', just in case one snuck in.)